File: leetcode-121-Best_Time_to_Buy_and_Sell_Stock/leetcode-121-Best_Time_to_Buy_and_Sell_Stock.cpp
#include <iostream>
#include <cstdlib>
#include <vector>


int maxProfit(std::vector<int>& prices) {
    int profit = 0,min = 0;
    for(size_t i = 0; i<prices.size(); ++i){
        if(i == 0 || prices[i]<min){
            min = prices[i];
        } else {
            continue;
        }
        for(size_t j = i+1; j<prices.size(); ++j){
            if(prices[j]<prices[i] || prices[j]==prices[i])
                continue;
            if(profit<prices[j]-prices[i])
            {
                profit = prices[j]-prices[i];
            }
        }
    }
    return profit;
}


int main(){
    std::vector<int> vec = {7,1,5,3,6,4};

    int res = maxProfit(vec);

    std::cout << res << std::endl;

    return EXIT_SUCCESS;
}

File: leetcode-121-Best_Time_to_Buy_and_Sell_Stock/leetcode-121-Best_Time_to_Buy_and_Sell_Stock.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        min_val:int = prices[0]
        result:int = 0
        for price in prices:
            min_val = min(price, min_val)
            result = max((price-min_val), result)
        return result


if __name__ == "__main__":
    prices: list[int] = [7,1,5,3,6,4]
    result = Solution().maxProfit(prices)
    print(result)
